Wabbajack.Lib/Downloaders/AbstractDownloadState.cs

```diff
@@ -7,21 +7,23 @@
 
 namespace Wabbajack.Lib.Downloaders
 {
-    public interface IMetaState
+    public abstract class AbstractMetaState : AbstractDownloadState
     {
-        string URL { get; }
-        string Name { get; }
-        string Author { get; }
-        string Version { get;}
-        string ImageURL { get; }
-        bool IsNSFW { get; }
-        string Description { get; }
+        public abstract string URL { get; }
+        public abstract string Name { get; set; }
+        public abstract string Author { get; set; }
+        public abstract string Version { get; set; }
+        public abstract string ImageURL { get; set; }
+        public abstract bool IsNSFW { get; set; }
+        public abstract string Description { get; set; }
+
+        public abstract Task<bool> LoadMetaData();
```

Comment:
I think I agree with @halgari a bit.  If it's just listing members/functions that should exist, then using an interface is much cleaner, as it better supports "multi-inheritence" of many interfaces.  Having it as a base class only is useful if it also implements something /w some logic.   As it stands, it's just declaring things that need to exist, so an interface is sufficient.